mlx/coverity/coverity_services.py

```diff
@@ -128,12 +129,19 @@ def validate_stream(self, stream):
     def validate_snapshot(self, snapshot):
         """Validate snapshot by retrieving the specified snapshot.
         When the request fails, the snapshot does not exist or the user does not have acces to it.
+        In this case a warning is logged and continues with the latest snapshot.
 
         Args:
             snapshot (str): The snapshot ID
         """
         url = f"{self.api_endpoint}/snapshots/{snapshot}"
-        self._request(url)
+        response = self.session.get(url)
+        if response.ok:
+            self.valid_snapshot = True
+            report_info(f"Snapshot ID {snapshot} is valid")
+        else:
+            report_warning(f"No snapshot found for ID {snapshot}; Continue with using the latest snapshot.", "")
+            self.valid_snapshot = False
```

Comment:
Instead of adding this attribute `valid_snapshot`, I suggest returning `snapshot`, which is either `"last()"` or a valid snapshot ID.

Reply:
Isn't it better to make `self.snapshot` instead of returning it because the validation is done before `get_defects` and we do not want to use 2 times the same request, right?

Reply:
I suggest returning a valid `snapshot`, and storing it in `SphinxCoverityConnector`.

`self.snapshot = self.coverity_service.validate_snapshot(self.snapshot)`

This value will later be used by `self.coverity_service.get_defects(self.stream, node["filters"], column_names, self.snapshot)`

Does this make sense?

Reply:
Resolved in f9b649189f2c8945b88514c4c54f1562a97ffe9e